internal/stacks/stacks_service.go

```diff
@@ -0,0 +1,196 @@
+package stacks
+
+import (
+	"context"
+	"fmt"
+	"github.com/kotalco/community-api/pkg/errors"
+	"github.com/kotalco/community-api/pkg/k8s"
+	"github.com/kotalco/community-api/pkg/logger"
+	stacksv1alpha1 "github.com/kotalco/kotal/apis/stacks/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type stacksService struct{}
+
+type IService interface {
+	Get(types.NamespacedName) (*stacksv1alpha1.Node, *errors.RestErr)
+	Create(dto *StacksDto) (node *stacksv1alpha1.Node, err *errors.RestErr)
+	List(namespace string) (*stacksv1alpha1.NodeList, *errors.RestErr)
+	Count(namespace string) (*int, *errors.RestErr)
+	Delete(node *stacksv1alpha1.Node) *errors.RestErr
+	Update(dto *StacksDto, node *stacksv1alpha1.Node) (*stacksv1alpha1.Node, *errors.RestErr)
+}
+
+var (
+	k8sClient = k8s.NewClientService()
+)
+
+func NewStacksService() IService {
+	return stacksService{}
+}
+
+// Create creates stacks node from spec
+func (service stacksService) Create(dto *StacksDto) (*stacksv1alpha1.Node, *errors.RestErr) {
+	node := &stacksv1alpha1.Node{
+		ObjectMeta: dto.ObjectMetaFromMetadataDto(),
+		Spec: stacksv1alpha1.NodeSpec{
+			Network:                  dto.Network,
+			Image:                    dto.Image,
+			NodePrivateKeySecretName: dto.NodePrivateKeySecretName,
```

Comment:
NodePrivateKeySecretName is not required during creation.
Nodes get assigned a random identifier when no private key is given.
private key -> public key -> node id.